### ISSUE TITLE
Make autogenerated BlazorComponent views partial class

### ISSUE THREAD

**Issue by uazo** [aspnet/AspNetCore#5487]
I would like to split the Razor c # + html template from the purely functional code of the view and I tried to make every generated class contained in "BlazorRazorComponents.g.cs" with the partial modifier (simply adding it in "Microsoft.AspNetCore.Blazor.Razor. Extensions \ BlazorRazorEngine.cs ").

So my views are composed in

nomeview.cshtml       <- html layout
nomeview.cshtml.cs   <- functional code

The compilation works without problems.
The problem is Visual Studio is unable to "read" the contents of the code in cshtml.cs, showing the error CS1061 'Class' does not contain a definition for 'Method'.

Now, adding the directive

`@inherits <name of the same class>`

instead Visual Studio works correctly, while, obviously, the compilation is not successful because the autogenerated code becomes

`public partial class <classname>: <classname> <- the same`

But if the code remain

`public partial class <classname>: BlazorComponent`

both the compilation and Visual Studio would work.

Is it possible to make that if implements the same class, BlazorComponent is kept as a base class?

**Comment by danroth27** [comment]
Depends on #4065

**Comment by springy76** [comment]
Such partial classes would also allow to specify generic constraints easily when using `@typeparam` https://github.com/aspnet/AspNetCore/issues/5603

**Comment by mkArtakMSFT** [comment]
@ajaybhargavb  should this be resolved as `Done` ?

**Comment by ajaybhargavb** [comment]
Yeah. I'll file a new issue to track addressing feedback and fixing bugs.

**Comment by ajaybhargavb** [comment]
Filed https://github.com/aspnet/AspNetCore/issues/14646
